crates/net/network/src/network.rs

```diff
@@ -135,10 +135,10 @@ impl NetworkHandle {
         self.send_message(NetworkHandleMessage::AnnounceBlock(block, hash))
     }
 
-    /// Sends a message to the [`NetworkManager`](crate::NetworkManager) to add a peer to the known
-    /// set
-    pub fn add_peer(&self, peer: PeerId, addr: SocketAddr) {
-        let _ = self.inner.to_manager_tx.send(NetworkHandleMessage::AddPeerAddress(peer, addr));
+    /// Sends a message to the [`NetworkManager`](crate::NetworkManager) to add or update
+    /// a peer to the known set
+    pub fn add_or_update_peer(&self, peer: PeerId, kind: PeerKind, addr: SocketAddr) {
```

Comment:
I'd like to kee `add_peer`, so I think we need 3 different functions

1. add_peer
2. add_trusted_peer
3. add_peer_kind

where 1),2) simply call 3) which is currently called `add_or_update_peer`

Reply:
Fixed in the last commit. Do you want `add_peer_kind` to be private?

Reply:
let's make it private